File: java/ucotron-sdk/src/main/java/com/ucotron/sdk/types/NeighborResponse.java
package com.ucotron.sdk.types;

/**
 * A neighbor node connected via an edge.
 */
public class NeighborResponse {
    private long node_id;
    private String content;
    private String edge_type;
    private float weight;

    public long getNodeId() { return node_id; }
    public String getContent() { return content; }
    public String getEdgeType() { return edge_type; }
    public float getWeight() { return weight; }
}

File: php/src/Types/NeighborResponse.php
<?php

declare(strict_types=1);

namespace Ucotron\Sdk\Types;

class NeighborResponse
{
    public function __construct(
        public readonly string $id,
        public readonly string $name,
        public readonly string $edge_type,
        public readonly float $weight,
    ) {}

    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            name: $data['name'] ?? '',
            edge_type: $data['edge_type'] ?? '',
            weight: (float) ($data['weight'] ?? 0.0),
        );
    }
}
